transactions/tpu-transaction-executor.ts

```diff
@@ -0,0 +1,56 @@
+import {
+    BlockhashWithExpiryBlockHeight,
+    Keypair,
+    VersionedTransaction
+} from '@solana/web3.js';
+import { AxiosError } from 'axios';
+
+import { TransactionExecutor } from './transaction-executor.interface';
+import { load, DataType, open, close } from 'ffi-rs';
+import bs58 from 'bs58';
+import { logger } from '../helpers';
+
+export class TpuTransactionExecutor implements TransactionExecutor {
+
+    constructor() { }
+
+    public async executeAndConfirm(
+        transaction: VersionedTransaction,
+        payer: Keypair,
+        latestBlockhash: BlockhashWithExpiryBlockHeight,
+        simulate: boolean
+    ): Promise<{ confirmed: boolean; signature?: string }> {
+        const serializedTransaction = transaction.serialize();
+        const signatureBase58 = bs58.encode(transaction.signatures[0]);
+        let result = false
+        try {
+            open({
+                library: 'tpu_client', // key
+                path: "/Users/kasiopea/dev/rust/tpu-sol-test/target/aarch64-apple-darwin/release/libtpu_client.dylib" // path
```

Comment:
???

Reply:
I've been researching how to call the TPU as people say transactions are faster, I've tried using 
[](https://github.com/lmvdz/tpu-client) but it doesn't work because solana migrated to quic, so the only way I've had to use it from TS is creating a mini library in rust and doing binding to ts using ffi-rs, this is the code
[](https://github.com/nikola43/tpu_client)

I forgot add compiled library and add it to project and leave my local path, my idea is build for mac, windows and linux for bot be compatible with any OS